include/fix8/sessionwrapper.hpp

```diff
@@ -646,10 +646,39 @@ class SessionInstance : public SessionInstanceBase
 	/// Dtor.
 	virtual ~SessionInstance ()
 	{
-		delete _session;
-		_session = nullptr;
-		delete _sock;
-		_sock = nullptr;
+		try
+		{
+			if (_psc != nullptr)
+			{
+				_psc->stop();
+				delete _psc;
+				_psc = nullptr;
+			}
+		}
+		catch (...)
+		{
+			std::cerr << "~SessionInstance - psc exception during cleanup" << std::endl;;
```

Comment:
Printing a message to `cerr` is not suitable here. You need to send an error message to one of the loggers, such as

    this->_session->log(e.what(), Logger::Error);